mbuild/compound.py

```diff
@@ -1457,6 +1442,69 @@ def is_independent(self):
                         return False
             return True
 
+    def check_for_overlap(self, excluded_bond_depth, minimum_distance=0.10):
+        """Check if a compound contains overlapping particles.
+
+        Parameters:
+        -----------
+        excluded_bond_depth : int, required
+            The depth of bonded neighbors to exclude from overlap check.
+            see Compound.direct_bonds()
+        minimum_distance : float, default=0.10
+            Distance (in nanometers) used as the threshold in
+            determining if a pair of particles overlap.
+
+        Notes:
+        ------
+        If `minimum_distance` is set larger than existing bond lengths,
+        adjust the `excluded_bond_depth` parameter to excluded directly
+        bonded neighbors from overlap checks.
+
+        See Also:
+        ---------
+        mbuild.Compound.direct_bonds()
+
+        Returns:
+        --------
+        overlapping_particles : list of lists
+            A list of particle pairs that were found within minimum_distance.
+        """
+        if excluded_bond_depth < 0 or not isinstance(excluded_bond_depth, int):
+            raise ValueError("`excluded_bond_depth must be an integer >= 0.")
+
+        if self.box:
+            min_length = np.min(self.box.lengths)
+        else:
+            min_length = np.min(self.get_boundingbox().lengths)
+
+        if minimum_distance >= min_length / 2:
+            raise ValueError(
+                "The minimum distance chosen is greater than or equal to "
+                "half of the box length."
+            )
+
+        freud = import_("freud")
+        moved_positions, freud_box = self.to_freud()
+        aq = freud.locality.AABBQuery(freud_box, moved_positions)
+        aq_query = aq.query(
+            query_points=moved_positions,
+            query_args=dict(r_min=0.0, r_max=minimum_distance, exclude_ii=True),
+        )
+        nlist = aq_query.toNeighborList()
+        # nlist contains each pair twice, get the set
+        pairs_set = set([tuple(sorted((i, j))) for i, j in nlist])
+        all_particles = [p for p in self.particles()]
+        overlapping_particles = []
+        for i, j in pairs_set:
+            # Exclude bonded neighbors that are within min distance
+            if excluded_bond_depth > 0:
+                i_bonds = all_particles[i].direct_bonds(graph_depth=excluded_bond_depth)
+                if all_particles[j] not in i_bonds:
+                    overlapping_particles.append([i, j])
```

Comment:
I think it's better to use a tuple here, instead of a list. Should be slightly more memory efficient since it's a fixed length.
```suggestion
                    overlapping_particles.append((i, j))
```